lib/Db/FileCollection.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Files_antivirus
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Viktar Dubiniuk <dubiniuk@owncloud.com>
+ *
+ * @copyright Viktar Dubiniuk 2019
+ * @license AGPL-3.0
+ */
+
+namespace OCA\Files_Antivirus\Db;
+
+use Doctrine\DBAL\Platforms\MySqlPlatform;
+use OCP\IDBConnection;
+
+class FileCollection {
+	/** @var IDBConnection */
+	private $dbConnection;
+
+	/**
+	 * FileCollection constructor.
+	 *
+	 * @param IDBConnection $dbConnection
+	 */
+	public function __construct(IDBConnection $dbConnection) {
+		$this->dbConnection = $dbConnection;
+	}
+
+	/**
+	 * @param int $fileSizeLimit
+	 *
+	 * @return \Doctrine\DBAL\Driver\Statement|int
+	 */
+	public function getCollection($fileSizeLimit) {
+		$dirMimeTypeId = $this->getDirectoryMimeTypeId();
+		$qb = $this->dbConnection->getQueryBuilder();
+		if ($this->dbConnection->getDatabasePlatform() instanceof MySqlPlatform) {
```

Comment:
any better alternative? I'd prefer not to deal with DB specific stuff in any app if possible.

Reply:
no alternative until the core mounts are changed. I need join strings and it is 'CONCAT' for MySQL only. For all other DB engines it will be `||`